### ISSUE TITLE
[Bug]:  Netclient fresh install suddenly opening ports on 443? This looks highly suspicous?

### ISSUE THREAD

**Issue by cocoonkid** [gravitl/netmaker#3262]
### What happened?

![image](https://github.com/user-attachments/assets/58fd98c8-9e19-4eac-8a61-8291d4408aff)

Not part fo my actual netclient network.

![image](https://github.com/user-attachments/assets/7e43253c-a1ff-4cdd-9891-50d0ba976a9d)

What network is this ?

### Version

v0.26.0

### Contributing guidelines

- [X] Yes, I did.

**Comment by cocoonkid** [comment]
okay I read the changelog. 

Reduced Firewall Requirements To One Single Port (443 udp/tcp)

soooo.  the whole world is running https services on this port. 
I do not agree this is a good idea :-)

**Comment by cocoonkid** [comment]
I sugges to change this. Many automations might try to deploy to 443  a few seconds or minutes later. 

Or users sometime else and then they need to find and change the netclient port  (which does not have a setting to even change it ?)

Also so many firewalls are akready preset  to the netmaker ports. They all have to change now...

I love the product and I love you guys but please think this over 🙂 We have so many ports. Also such a change should really be announced some time upfront. 

51821 was really fine. and only need one port ist also amazing. 

But should stick to what has been preset. IMHO 🙂

**Comment by cocoonkid** [comment]
And I suggest adding the IP that it is going to connect to (in my case 10.10.2.18) to the server UI so it is immediately evident.

**Comment by abhishek9686** [comment]
> I sugges to change this. Many automations might try to deploy to 443 a few seconds or minutes later.
> 
> Or users sometime else and then they need to find and change the netclient port (which does not have a setting to even change it ?)
> 
> Also so many firewalls are akready preset to the netmaker ports. They all have to change now...
> 
> I love the product and I love you guys but please think this over 🙂 We have so many ports. Also such a change should really be announced some time upfront.
> 
> 51821 was really fine. and only need one port ist also amazing.
> 
> But should stick to what has been preset. IMHO 🙂

This won't affect existing netclients only for new clients, netclient also has the option to specify port while joining the network or you can also change it in the UI
doesn't really affect your existing services.
 This is mainly done for environments with hard restrictions for opening many ports and wanted to improve the first-time experience for people deploying netclient in their environment because in most environments outbound 443 will be allowed.

**Comment by abhishek9686** [comment]
Yes, I know this is an enterprise product and our enterprise customers requested that we reduce the port requirements to a single port preferably to 443. We have built the client to be configurable, and all you need to do is add your desired port in the cmd argument while joining the network. 
`netclient join -t <key> --static-port -p 51821`

If users are facing any issues with this change we are happy to revert the change

**Comment by cocoonkid** [comment]
I fully understand and it's a good change but FORCING the 443 is kinda the issue. 
Keeping the standard for now and announcing "we now use just one port which is awesome, with version 3.2.0 we will make it the default" would habe been a tad more user friendly. Then even people like me have time to RTFM and adjust!

Thank you for taking the time to reply. 
My aplogy if I came across in a bad way/mood. 
English is not my native language. 

I found netclient list now has more output too which broke all my parsers. That's okay :-) 

cheers

**Comment by christian-schlichtherle** [comment]
Using port 443 by default is a bad choice for all the reasons @cocoonkid has explained already. I was also bitten by this regression and therefore found this ticket. The default port should be reverted back to what it was before: 51821 - 51830 (the first free port).

**Comment by cocoonkid** [comment]
Looks like the enshitification of netmaker has officially begun. I migrated all my nodes to alternatives.

**Comment by abhishek9686** [comment]
@christian-schlichtherle in the upcoming release we have made changes to use only 443/udp  by default and it will fallback to 51821 if it is unavailable

**Comment by christian-schlichtherle** [comment]
@abhishek9686 I respect your efforts, but this doesn't help at all. The default should be left 51821 - 51830. 443 should only be used if explicitly configured, so opt-in.
